src/wp-includes/class-wp-query.php

```diff
@@ -4714,11 +4714,20 @@ public function generate_postdata( $post ) {
 
 		$authordata = get_userdata( $post->post_author );
 
-		$currentday   = mysql2date( 'd.m.y', $post->post_date, false );
-		$currentmonth = mysql2date( 'm', $post->post_date, false );
-		$numpages     = 1;
-		$multipage    = 0;
-		$page         = $this->get( 'page' );
+		$post_date = $post->post_date;
+		if ( ! empty( $post_date ) && '0000-00-00 00:00:00' !== $post_date ) {
+			list( $year, $month, $day ) = explode( '-', substr( $post_date, 0, 10 ) );
+
+			$currentday   = sprintf( '%s.%s.%s', $day, $month, substr( $year, -2 ) );
```

Comment:
If using `sprintf()`, since we expect zero-padded two-digit integers, I think it's prudent to express that in format. That would be `%02d.%02d.%02d` on top of my head.

Reply:
I don't think this one was needed. I can follow up with another commit, if important.

Reply:
Not important, cheers.